// @secondhand/shared/src/index.ts
export type UserRole = "admin" | "user";

export const BUILTIN_QUICK_REPLIES = [
  "可小刀吗？",
  "还在吗？",
  "最低多少？",
] as const;

export interface User {
  id: string;
  username: string;
  password: string;
  role: UserRole;
  status: "active" | "pending_review" | "rejected";
  reviewNote?: string;
  quickReplies?: string[];
  createdAt: string;
}

export type PublicUser = Omit<User, "password">;

export interface QuickRepliesPayload {
  builtin: string[];
  custom: string[];
}

export type ProductStatus = "pending" | "approved" | "rejected" | "offline" | "sold";

export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  category: "digital" | "book" | "daily" | "ticket" | "other";
  images: string[];
  campus: string;
  brand?: string;
  model?: string;
  memory?: string;
  latitude?: number;
  longitude?: number;
  status: ProductStatus;
  rejectionReason?: string;
  sellerId: string;
  sellerName: string;
  createdAt: string;
  updatedAt: string;
}

export interface RecommendationFeed {
  items: Product[];
  reason: string;
}

export interface LoginBody {
  username: string;
  password: string;
}

export interface RegisterBody {
  username: string;
  password: string;
  role: UserRole;
}

export interface CreateProductBody {
  title: string;
  description: string;
  price: number;
  category: "digital" | "book" | "daily" | "ticket" | "other";
  images: string[];
  campus: string;
  brand?: string;
  model?: string;
  memory?: string;
  latitude?: number;
  longitude?: number;
}

export interface ProductMessage {
  id: string;
  productId: string;
  fromUserId: string;
  fromUsername: string;
  toUserId?: string;
  toUsername?: string;
  content: string;
  createdAt: string;
}

export type OrderStatus = "in_progress" | "completed" | "cancelled";

export interface Order {
  id: string;
  productId: string;
  productTitle: string;
  buyerId: string;
  buyerName: string;
  sellerId: string;
  sellerName: string;
  price: number;
  status: OrderStatus;
  buyerConfirmed?: boolean;
  sellerConfirmed?: boolean;
  createdAt: string;
  rating?: number;
  ratedAt?: string;
  reviewText?: string;
  reviewImages?: string[];
}

export interface OrderReviewBody {
  rating: number;
  reviewText?: string;
  reviewImages?: string[];
}

export interface SellerPublicProfile {
  id: string;
  username: string;
  trustScore: number;
  school: string;
  publishedCount: number;
  completedOrderCount: number;
  products: Product[];
}
